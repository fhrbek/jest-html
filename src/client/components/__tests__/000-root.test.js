/* eslint-env jest */
/* eslint-disable global-require, import/newline-after-import */
import React from 'react';
import renderer from 'react-test-renderer';
import Root from '../000-root';

// https://github.com/facebook/react/issues/7386#issuecomment-238091398
jest.mock('react-dom');
jest.mock('react-router', () => ({
  Switch: require('./mockComponent')('Switch'),
}));
jest.mock('react-router-dom', () => ({
  BrowserRouter: require('./mockComponent')('BrowserRouter'),
  Route: require('./mockComponent')('Route'),
}));
jest.mock('../010-app', () => require('./mockComponent')('App'));
jest.mock('../200-largeMessage', () =>
  require('./mockComponent')('LargeMessage')
);

describe('Root', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Root />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
