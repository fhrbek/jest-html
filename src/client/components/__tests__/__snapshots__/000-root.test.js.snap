// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root renders correctly 1`] = `
<div
  dataMockType="BrowserRouter"
>
  <div
    style={
      Object {
        "color": "blue",
        "fontWeight": "bold",
      }
    }
  >
    BrowserRouter
  </div>
  <div
    dataMockType="Switch"
  >
    <div
      style={
        Object {
          "color": "blue",
          "fontWeight": "bold",
        }
      }
    >
      Switch
    </div>
    <div
      component={[Function]}
      dataMockType="Route"
      exact={true}
      path="/"
    >
      <div
        style={
          Object {
            "color": "blue",
            "fontWeight": "bold",
          }
        }
      >
        Route
      </div>
    </div>
    <div
      component={[Function]}
      dataMockType="Route"
      path="/suite/*"
    >
      <div
        style={
          Object {
            "color": "blue",
            "fontWeight": "bold",
          }
        }
      >
        Route
      </div>
    </div>
    <div
      component={[Function]}
      dataMockType="Route"
      path="/folder/*"
    >
      <div
        style={
          Object {
            "color": "blue",
            "fontWeight": "bold",
          }
        }
      >
        Route
      </div>
    </div>
    <div
      dataMockType="Route"
      render={[Function]}
    >
      <div
        style={
          Object {
            "color": "blue",
            "fontWeight": "bold",
          }
        }
      >
        Route
      </div>
    </div>
  </div>
</div>
------------HTML PREVIEW---------------
<div dataMockType="BrowserRouter">
  <div style="color:blue;font-weight:bold;">
    BrowserRouter
  </div>
  <div dataMockType="Switch">
    <div style="color:blue;font-weight:bold;">
      Switch
    </div>
    <div
      dataMockType="Route"
      exact
      path="/"
    >
      <div style="color:blue;font-weight:bold;">
        Route
      </div>
    </div>
    <div
      dataMockType="Route"
      path="/suite/*"
    >
      <div style="color:blue;font-weight:bold;">
        Route
      </div>
    </div>
    <div
      dataMockType="Route"
      path="/folder/*"
    >
      <div style="color:blue;font-weight:bold;">
        Route
      </div>
    </div>
    <div dataMockType="Route">
      <div style="color:blue;font-weight:bold;">
        Route
      </div>
    </div>
  </div>
</div>
`;
